crates/sparrow-syntax/src/syntax/fenl_type.rs

```diff
@@ -19,7 +19,11 @@ pub enum FenlType {
     /// A specific Arrow DataType.
     Concrete(DataType),
     /// A generic type with the given type variable.
-    Generic(TypeVariable),
+    TypeRef(TypeVariable),
+    /// A collection type with the given type variable(s).
+    ///
+    /// e.g. (Collection::Map, [TypeVariable("K"), TypeVariable("V")])
+    Collection(Collection, Vec<TypeVariable>),
```

Comment:
I think the arguments to the collection should be `FenlType`. Otherwise, we can't handle `Map<string, string>` (map from string to string), `Map<string, Map<i32>>` (nested maps), etc.

Reply:
https://github.com/kaskada-ai/kaskada/issues/494